DQM/EcalMonitorTasks/interface/TimingTask.h

```diff
@@ -19,10 +19,18 @@ namespace ecaldqm {
     void runOnRecHits(EcalRecHitCollection const&, Collections);
     void runOnUncalibRecHits(EcalUncalibratedRecHitCollection const&);
 
+    enum Constants {
+      nBXBins = 15
+    };
+
   private:
     void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+    void beginEvent(edm::Event const&, edm::EventSetup const&) override;
     void setParams(edm::ParameterSet const&) override;
 
+    int bxBinEdges_[nBXBins + 1];
```

Comment:
Hi @davidlange6 , thanks for the suggestion, but using std::array is inconsistent with the syntax used here: https://github.com/tanmaymudholkar/cmssw/blob/TimingVsBX_and_LaserTiming_91X/DQM/EcalMonitorTasks/src/TimingTask.cc#L65

I used this construction to remain consistent with another very similar plot that we've had for some time: https://github.com/tanmaymudholkar/cmssw/blob/TimingVsBX_and_LaserTiming_91X/DQM/EcalMonitorTasks/src/TrigPrimTask.cc#L79-L80

Perhaps I could try to replace both of them by std::vectors that read in the bin values from the config file? Do you know if that would work with std::upper_bound()? That might take me some time (~1 day) to test etc. Please advise.